src/types/type_param.rs

```diff
@@ -21,6 +21,9 @@ pub enum TypeParam {
     Type(TypeBound),
     /// Argument is a [TypeArg::USize].
     USize,
+    /// Argument is a [TypeArg::SimplePredicate] which is bounded by the size
+    /// specified here.
+    SimplePredicate(usize),
```

Comment:
Would it be possible to call this something like `BoundedUsize` which is more descriptive, even if it is implemented as a sum type?

Reply:
I'd like to maintain the analogy with predicates in the main type system, but I agree a more descriptive name would be better. Maybe `UnitPredicate` (and update simple predicate to that in the main code base too?)

Reply:
I was thinking the same (BoundedUSize) but I take your argument. Really I think "predicate" is the wrong word to start. We're calling it that because we use it to control branches, but IMHO predicate is inherently binary - true or false - Google gives me this:

> 2.
> LOGIC
> something which is affirmed or denied concerning an argument of a proposition

So perhaps go with predicate for now but a better word might be "selector" or "choice"? or wilder ideas like "Nway"

Reply:
I like `Choice`.

Reply:
Yeah I can make a new issue for that rename, and that can cover this too when it is done.

Reply:
Yeah that works - the yaml syntax could offer "bool" which maps to Usize(bound: 2)

Reply:
though I prefer then `BoundedU` as the only one, with the default being the maximum (`BoundedU(u64::MAX)`)

Reply:
Yes, I was meaning the Option to express the idea of defaulting to None, too

Reply:
I'd still approve of renaming `Predicate` to `Choice`, but if that can be made to work it sounds quite appealing.

Reply:
yeah I think that is a separate task in the main codebase